dotnet/src/AutoGen.Core/GroupChat/GroupChat.cs

```diff
@@ -140,37 +167,45 @@ public void AddInitializeMessage(IMessage message)
     }
 
     public async Task<IEnumerable<IMessage>> CallAsync(
-        IEnumerable<IMessage>? conversationWithName = null,
+        IEnumerable<IMessage>? chatHistroy = null,
         int maxRound = 10,
         CancellationToken ct = default)
     {
         var conversationHistory = new List<IMessage>();
-        if (conversationWithName != null)
+        conversationHistory.AddRange(this.initializeMessages);
+        if (chatHistroy != null)
         {
-            conversationHistory.AddRange(conversationWithName);
+            conversationHistory.AddRange(chatHistroy);
         }
+        var roundLeft = maxRound;
 
-        var lastSpeaker = conversationHistory.LastOrDefault()?.From switch
-        {
-            null => this.agents.First(),
-            _ => this.agents.FirstOrDefault(x => x.Name == conversationHistory.Last().From) ?? throw new Exception("The agent is not in the group chat"),
-        };
-        var round = 0;
-        while (round < maxRound)
+        while (roundLeft > 0)
         {
-            var currentSpeaker = await this.SelectNextSpeakerAsync(lastSpeaker, conversationHistory);
-            var processedConversation = this.ProcessConversationForAgent(this.initializeMessages, conversationHistory);
-            var result = await currentSpeaker.GenerateReplyAsync(processedConversation) ?? throw new Exception("No result is returned.");
+            var orchestratorContext = new OrchestrationContext
+            {
+                Candidates = this.agents,
+                ChatHistory = conversationHistory,
+            };
+            var nextSpeaker = await this.orchestrator.GetNextSpeakerAsync(orchestratorContext, ct);
+            if (nextSpeaker == null)
+            {
+                break;
+            }
+
+            var result = await nextSpeaker.GenerateReplyAsync(conversationHistory, cancellationToken: ct);
             conversationHistory.Add(result);
 
-            // if message is terminate message, then terminate the conversation
-            if (result?.IsGroupChatTerminateMessage() ?? false)
+            if (result.IsGroupChatTerminateMessage())
             {
-                break;
+                return conversationHistory;
             }
 
-            lastSpeaker = currentSpeaker;
-            round++;
+            roundLeft--;
+
+            if (roundLeft <= 0)
+            {
+                break;
+            }
```

Comment:
Is this check redundant since the while loop checks `roundleft > 0 ` ?

Reply:
Nope